org.eclipse.xtext/src/org/eclipse/xtext/validation/AbstractDeclarativeValidator.java

```diff
@@ -133,10 +139,7 @@ public void invoke(State state) {
 		}
 		
 		protected void handleInvocationTargetException(Throwable targetException, State state) {
-			// ignore GuardException, check is just not evaluated if guard is false
-			// ignore NullPointerException, as not having to check for NPEs all the time is a convenience feature
-			if (!(targetException instanceof GuardException) && !(targetException instanceof NullPointerException))
-				Exceptions.throwUncheckedException(targetException);
+			instance.handleExceptionDuringValidation(targetException);
```

Comment:
I'm not sure if this should  be `instance.handleException` or `AbstractDeclarativeValidator.this.handleException`.

Line of thought:
One can compose library validators as composed checks. Usually you would not want to override all these to make the NPEs being rethrown, but only your "root"-validator. What's your take on this?